test/unit/org/apache/cassandra/batchlog/BatchlogEndpointFilterTest.java

```diff
@@ -103,12 +154,13 @@ public void shouldSelectLastHostsFromLastNonLocalRacks() throws UnknownHostExcep
     @Test
     public void shouldSelectHostFromLocal() throws UnknownHostException
     {
+        DatabaseDescriptor.setBatchlogEndpointStrategy(Config.BatchlogEndpointStrategy.random_remote);
         Multimap<String, InetAddressAndPort> endpoints = ImmutableMultimap.<String, InetAddressAndPort> builder()
                 .put(LOCAL, InetAddressAndPort.getByName("0"))
                 .put(LOCAL, InetAddressAndPort.getByName("00"))
                 .put("1", InetAddressAndPort.getByName("1"))
                 .build();
-        Collection<InetAddressAndPort> result = filterBatchlogEndpoints(false, endpoints);
+        Collection<InetAddressAndPort> result = filterBatchlogEndpointsRandom(false, endpoints);
         assertThat(result.size()).isEqualTo(2);
         assertTrue(result.contains(InetAddressAndPort.getByName("1")));
         assertTrue(result.contains(InetAddressAndPort.getByName("0")));
```

Comment:
does results always contain the address `0` ? What about `00`; can this be in the result ?

Reply:
okay. I think I understand now but please confirm for me. I saw below the filterBatchlogEndpointsRandom function is slightly modified to always return the last one and that's why 0 will not be in the results